### ISSUE TITLE
Please add copyright and license notice to README.md

### ISSUE THREAD

**Issue by jawnsy** [minio/minio-go#870]
The project contains an Apache 2.0 license file, however, it does not follow the directions in the appendix of the license: https://github.com/minio/minio-go/blob/master/LICENSE#L179-L202

For clarity, the README should "apply" the license as directed, including copyright information. Since the code was written primarily by the Minio team, it's probably easiest to indicate Minio, Inc as the copyright owner.

**Comment by harshavardhana** [comment]
> The project contains an Apache 2.0 license file, however, it does not follow the directions in the appendix of the license: https://github.com/minio/minio-go/blob/master/LICENSE#L179-L202
> 
> For clarity, the README should "apply" the license as directed, including copyright information. Since the code was written primarily by the Minio team, it's probably easiest to indicate Minio, Inc as the copyright owner.

Each file already carries there is no such indication in the appendix regarding README.md @jawnsy

**Comment by jawnsy** [comment]
@harshavardhana Ah, sorry, I missed that as I did a search for the "Copyright" string. I think the word "Copyright" is important in some regions; (c) is not sufficient. Debian packagers typically run something like `grep -ir copyright .` which would not catch the notices as they exist today.

I'd suggest following the form provided in the appendix: `Copyright [yyyy] [name of copyright owner]` (though unfortunately this would require updating the headers on all files)

**Comment by harshavardhana** [comment]
While that might be true that is a weird way to handle this in Debian. 

We are following a quite common format here. There is no reason to grep when the project src directory has LICENSE file.

**Comment by jawnsy** [comment]
@harshavardhana Copyright is required to assert a license, so it is tracked closely in Debian. Packages are routinely [rejected by the ftp-master team](https://ftp-master.debian.org/REJECT-FAQ.html) because of missing or incomplete copyright information. It's true that not all users of, or contributors to, your project will be as picky in this regard, but why not make it easier for those that care about it? Particularly given the boilerplate statement in the Apache license already has an easily-searchable statement.

As mentioned, I think that the use of "(c)" is also a bit of a grey area, because [United States Code, Title 17, Section 401(b)(1)](https://www.law.cornell.edu/uscode/text/17/401#b_1) states the options as: _the symbol © (the letter C in a circle), or the word “Copyright”, or the abbreviation “Copr.”_ That said, I'm not a lawyer, so you should consider retaining one or following the template that was approved by the Apache Software Foundation's lawyers. Given the minimal cost of the latter (applying the statement as specified in the license text itself), I don't understand your objection here?

PS: I know you probably don't mean it this way, but some of your comments sound unwelcoming. Not all open source projects need/want outside contributors, however if Minio is interested in building a community, I suggest you consider how your messages may read to others. Your comment about the use of `grep -ir copyright .` to find copyright information being "weird" sounds quite strongly worded, and from my experience as a former Debian contributor, this works quite well for the great majority of projects.

**Comment by harshavardhana** [comment]
> @harshavardhana Copyright is required to assert a license, so it is tracked closely in Debian. Packages are routinely rejected by the ftp-master team because of missing or incomplete copyright information. It's true that not all users of, or contributors to, your project will be as picky in this regard, but why not make it easier for those that care about it? Particularly given the boilerplate statement in the Apache license already has an easily-searchable statement.

At the time when we followed this format we agreed that this is sufficient and that its common enough. Since top level folder already carries the right LICENSE header, simpler form as below was considered to be fine and was assumed (from other projects) common enough in practice. 

```
[project] (C) [date] [copyright owner]
```

We can potentially change this since in GNU projects we used to follow a different format

```
[project]
Copyright (C) [date] [copyright owner]
```

or simply

```
Copyright (C) [date] [copyright owner]
```

> PS: I know you probably don't mean it this way, but some of your comments sound unwelcoming. Not all open source projects need/want outside contributors, however if Minio is interested in building a community, I suggest you consider how your messages may read to others. Your comment about the use of grep -ir copyright . to find copyright information being "weird" sounds quite strongly worded, and from my experience as a former Debian contributor, this works quite well for the great majority of projects.

It wasn't meant to be unwelcoming @jawnsy **we always welcome external contributors**.

As a fellow programmer i find it unusual that `grep -ir` is relied on to assert licenses.  If you look at projects like https://github.com/aws/aws-sdk-go/blob/master/service/s3/host_style_bucket.go do not even have a copyright header and they are perfectly packaged fine under https://packages.debian.org/sid/golang-github-aws-aws-sdk-go-dev

Similarly https://packages.debian.org/sid/golang-github-minio-minio-go-dev is already packaged as well and managed here https://anonscm.debian.org/git/pkg-go/packages/golang-github-minio-minio-go.git/tree/debian/copyright

In conclusion lets discuss this further with others as well and fix it.

**Comment by harshavardhana** [comment]
Here are the requirements from Apache 2.0 point of view

- Source headers should have a copyright under the following form 
```
Copyright [yyyy] [copyright owner]
```

- Top level of the project should have a NOTICE file in the following form
```
[PRODUCT_NAME]
Copyright [yyyy-yyyy] [OWNER]

This product includes software developed at [OWNER]
```

- LICENSE and NOTICE file preferably not of the form `.txt` 

Refer http://www.apache.org/legal/src-headers.html

**Comment by jawnsy** [comment]
@harshavardhana Right, the Apache license requirements say to use the word "Copyright"; it does not say that "(c)" is an acceptable alternative, nor do any legal opinions I've read. Do you have a link to one that says otherwise?

The `golang-github-aws-aws-sdk-go-dev` package you refer to has a [NOTICE.txt file](https://github.com/aws/aws-sdk-go/blob/master/NOTICE.txt) that contains the following:

```
Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved. 
Copyright 2014-2015 Stripe, Inc.
```

This would be found by the grep command identified. The "All Rights Reserved" part seems a bit dubious as well, as the point of open source is that the company is providing a perpetual license of some of those rights to others, but hey, I'm not a lawyer, and I digress.

All of this software is packaged manually (by Debian Developers), and reviewed manually (by ftp-master), and it's possible that they share your opinion regarding the acceptability of (c) in copyright statements, or that they simply missed it in the initial review, but that isn't sufficient justification against trying to do better. Additionally, international copyright law (following the Berne Convention) does not _require_ such statements in order to have copyright protection, but they are helpful for the sake of clarity.

I've worked on hundreds, possibly thousands, of packages in Debian in the past, and found that `grep` command to be pretty helpful when composing `debian/copyright` files. It's not that the `grep` command is required to assert licenses, but that a copyright statement is, and the grep command makes it easy to quickly find copyright statements. It's a good practice to check that because code is sometimes embedded from elsewhere and may carry a different license/copyright statement.

It might have been better for me to use [licensecheck](https://packages.debian.org/sid/licensecheck), but again, the simple grep command has usually worked pretty well for me in the past.

Also, the statement you linked says: "This section refers only to works submitted directly to the ASF by the copyright owner or owner's agent." which is not applicable to Minio. I don't think those are the requirements for Apache-licensed projects in general, but specifically for those managed by the ASF itself.

**Comment by harshavardhana** [comment]
The PR I referenced already took care of this. We will have to do this across our projects under minio org. ( Since the agreed previous pattern is followed quite commonly everywhere )

Again thanks for bringing this to our notice, we will fix it appropriately based on the Apache 2.0 convention, move to a more friendlier convention.

**Comment by harshavardhana** [comment]
So we agreed to change the license headers across minio/* projects to follow following format. 

```
[project title]
Copyright (C) [yyyy-yyyy] [owner]
```
